gui/src/components/Taipy/Chart.tsx

```diff
@@ -367,15 +368,16 @@ const Chart = (props: ChartProp) => {
     const onRelayout = useCallback(
         (eventData: PlotRelayoutEvent) => {
             onRangeChange && dispatch(createSendActionNameAction(id, { action: onRangeChange, ...eventData }));
-            if (decimator) {
+            if (decimator && !config.types.includes("scatter3d")) {
                 const backCols = Object.keys(config.columns).map((col) => config.columns[col].dfid);
                 const eventDataKey = Object.keys(eventData).map(v => v + "=" + eventData[v as keyof typeof eventData]).join("-");
                 dataKey.current = backCols.join("-") + (decimator ? `--${decimator}` : "") + "--" + eventDataKey;
                 const decimatorPayload = {
                     width: plotRef.current?.clientWidth,
                     height: plotRef.current?.clientHeight,
                     xAxis: config.traces.length && config.traces[0].length && config.traces[0][0] && config.columns[config.traces[0][0]].dfid,
-                    yAxis: config.traces.length == 1 && config.traces[0].length > 1 && config.columns[config.traces[0][1]] && config.columns[config.traces[0][1]].dfid,
+                    yAxis: config.traces.length > 0 && config.traces[0].length > 1 && config.traces[0][1] && config.columns[config.traces[0][1]] && config.columns[config.traces[0][1]].dfid,
+                    zAxis: config.traces.length > 0 && config.traces[0].length > 2 && config.traces[0][2] && config.columns[config.traces[0][2]] && config.columns[config.traces[0][2]].dfid,
```

Comment:
I guess we should extract this as a function 
`const getAxis = (traces, columns, index) => traces.length && traces[0].length > index && traces[0][index] && columns[traces[0][index]] && columns[traces[0][index]].dfid`

Reply:
and maybe even the complete decimator payload creation ...

Reply:
Thanks!